File: src/main/java/dev/toastmc/client/module/ModuleManifest.java
package dev.toastmc.client.module;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface ModuleManifest {
    String label();
    String[] aliases() default {};
    String description() default "";
    String usage() default "";
    Category category() default Category.NONE;
    boolean hidden() default false;
    boolean persistent() default false;
    boolean enabled() default false;
    int key() default -1;
}

File: src/main/java/dev/toastmc/client/module/ModuleManifest.kt
package dev.toastmc.client.module

@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
annotation class ModuleManifest(val label: String, val aliases: Array<String> = [], val description: String = "", val usage: String = "", val category: Category = Category.NONE, val hidden: Boolean = false, val persistent: Boolean = false, val enabled: Boolean = false, val key: Int = -1) 